server/ai_live_video.go

```diff
@@ -138,5 +139,35 @@ func startControlPublish(control *url.URL, params aiRequestParams) {
 	}
 	params.node.LiveMu.Lock()
 	defer params.node.LiveMu.Unlock()
-	params.node.LivePipelines[stream] = &core.LivePipeline{ControlPub: controlPub}
+
+	ticker := time.NewTicker(10 * time.Second)
+	done := make(chan bool, 1)
+	stop := func() {
```

Comment:
Shouldn't this close the publisher too?

Reply:
The publisher is closed just before stop is invoked [here](https://github.com/livepeer/go-livepeer/pull/3299/files/9f4a9793c757f1c4945beb538b48d5772d6a6e9b#diff-88f3e5816c6c29e92773629b388dfc58ed59a3682ab624c7f3448cfda3106475R562) ... we could try to combine them but I am a little hesitant to change the logic outside the timer stuff.

How it works right now is a little lazy:
If _gateway_ closes the publish channel, then it immediately stops the timer 
If _orchestrator_ closes the publish channel, then the timer stops on the next tick when the write errors out (no need to close the publisher at this point since it's already closed)